core/src/com/mygdx/game/models/TerrainClass.java

```diff
@@ -0,0 +1,71 @@
+package com.mygdx.game.models;
+
+import com.badlogic.gdx.Gdx;
+import com.badlogic.gdx.files.FileHandle;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+
+/**
+ * A class that represents a tile in the game.
+ */
+public class TerrainClass implements Cloneable, Terrain {
+    private String type;
+    private int defense;
+    private TerrainState terrainState;
+
+    private TerrainClass(String type, int defense, TerrainState terrainState) {
+        this.type = type;
+        this.defense = defense;
+        this.terrainState = terrainState;
+    }
+
+    // Copy constructor
+    public TerrainClass(TerrainClass original) {
+        type = original.type; // Immutable
+        defense = original.defense; // Immutable
+        terrainState = original.terrainState; // Not sure if this is the proper way to copy enum
+    }
+
+    /**
+     * Returns the Terrain from the configuration file.
+     *
+     * @param filename the filename
+     * @return the Terrain from the configuration file
+     */
+    public static TerrainClass getTerrainFromConfig(String filename) {
+        FileHandle file = Gdx.files.internal(filename);
```

Comment:
I added a Congiguration getter class with a single method that gets a configuration file based on a path. Can you use this method instead. This way if we move the Config files around then we only need to make a change in one place instead of everywhere.